com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingManager.cs

```diff
@@ -114,6 +115,32 @@ internal void ReleaseRayTracingManager()
                 m_DiffuseDenoiser.Release();
         }
 
+        bool IsValidRayTracedMaterial(Material currentMaterial)
+        {
+            if (currentMaterial == null || currentMaterial.shader == null)
+                return false;
+
+            bool isValid;
+
+            // We use a cache, to speed up the case where materials are reused many times
+            int matId = currentMaterial.GetInstanceID();
```

Comment:
Hey, so I don't really like the fact that we have a cache that can grow indefinitely (or which are reset every frame). I am agree with @RemyUnity suggestion that it should used shader instead.

What about using this instead:

int shaderId = currentMaterial.shader.GetInstanceID();
if (m_ShaderValidityCache.TryGetValue(shaderId , out isValid))
                return isValid;
                
if (currentMaterial..FindPass("IndirectDXR", false)
   isValid = true;

return false;

if we found the pass "IndirectDXR" it mean the shaders IS HDRP AND support DXR :)
                
thougth?

Reply:
And we never clear the cache in this case (we shouldn't have a huge amount of shader)

Reply:
After further tests and discussion, I have followed the suggestion to move to a persistent shader-based cache (no clear at each frame anymore), however the actual test on the pass name didn't work, so we keep it as it was before. But since the cache hit rate is now extremely high, I removed the faster, dirtier test based on substrings.